src/robot.cpp

```diff
@@ -469,8 +471,20 @@ void Robot::setSpeed(dReal vx, dReal vy, dReal vw)
     if (abs(a) > aLimit) {
         a = copysign(aLimit, a);
         dReal new_v = cv + a * cfg->DeltaTime() * 2;
-        vx *= new_v / v;
-        vy *= new_v / v;
+        if (v > 0) {
+            vx *= new_v / v;
+            vy *= new_v / v;
+        } else {
+            // convert global to local
+            dReal k, angle;
+            angle = getDir(k);
```

Comment:
There is also a `getDir()` method without `k`. As far as I can see, you do not use `k`, so can you just remove `k`?

Reply:
I had looked for it and did not see it, I had totally missed it... woosh
I pushed a new commit removing the `k`